lib/jet/peer.js

```diff
@@ -70,8 +70,12 @@ var Peer = function (config) {
           that.rejectConnect(err)
         })
       } else {
-        that.connected = true
-        that.resolveConnect(that)
+        return that.peerId().then(function (peerId) {
+          that.id = peerId
+        }).catch(function () {}).then(function () {
+          that.connected = true
+          that.resolveConnect(that)
```

Comment:
On the client side, you could do this jsonrpc.js. It's not ideal because it's not actually part of jsonrpc but it's kind of like we're augmenting the protocol. In peer/jsonrpc.js, you'll have to remove the listener for the open event. In the listener for message, check this._isOpened. If it's false, then you can assume it's the first message containing your peer id. Set this._isOpened to true and resolve the connect deferred with the message that is {id: 'asdfafsd'}. Then in peer.js you can take that resolved value to set peer.id.